contrib/win32/win32compat/fileio.c

```diff
@@ -663,6 +726,11 @@ fileio_close(struct w32_io* pio)
 {
 	debug2("fileclose - pio:%p", pio);
 
+	if (WINHANDLE(pio) == 0 || WINHANDLE(pio) == INVALID_HANDLE_VALUE) {
```

Comment:
when will be have the invalid handle? can you add some comments in which scenario this may happen...